rbd/mirror_desc_status_test.go

```diff
@@ -0,0 +1,164 @@
+//go:build !nautilus && ceph_preview
+// +build !nautilus,ceph_preview
+
+// IMPORTANT - when removing ceph_preview from this file also delete
+// rbd/mirror_desc_status_test.go as it will no longer serve a purpose.
+
+package rbd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMirrorDescriptionJSON(t *testing.T) {
+	cases := []struct {
+		desc string
+		ok   bool
+	}{
+		{
+			desc: "",
+			ok:   false,
+		},
+		{
+			desc: "local image is primary",
+			ok:   false,
+		},
+		{
+			desc: "status not found",
+			ok:   false,
+		},
+		{
+			desc: "invalid {",
+			ok:   false,
+		},
+		{
+			desc: "} invalid {",
+			ok:   false,
+		},
+		{
+			desc: "phony, {}",
+			ok:   true,
+		},
+		{
+			desc: `replaying, {"bytes_per_second":0.0,"bytes_per_snapshot":0.0,"last_snapshot_bytes":0,"last_snapshot_sync_seconds":0,"remote_snapshot_timestamp":1678125999,"replay_state":"idle"}`,
+			ok:   true,
+		},
+		{
+			desc: "invalid-json, {:::...!}",
+			ok:   false,
+		},
+	}
+	for _, tcase := range cases {
+		t.Run("testParse", func(t *testing.T) {
+			var data map[string]interface{}
+			s := SiteMirrorImageStatus{
+				Description: tcase.desc,
+			}
+			err := s.UnmarshalDescriptionJSON(&data)
+			if tcase.ok {
+				assert.NoError(t, err)
```

Comment:
Perhaps.  Because UnmarshalDescriptionJSON is the "low level" version that allows you to try and unmarshal to any type you wan I didn't think that doing anything more than verifying the unmarshal worked was strictly necessary.  It also makes writing a table test harder because then I need a unique function for every unique json object. Let me know if you feel strongly enough and I'll add it, otherwise I'd pefer to skip it.

Reply:
Ok. I won't insist.